### DISCUSSION TITLE
[Feature request]: Purpose: Game Status Portal

### DISCUSSION THREAD

**Discussion by EliasTheGrandMasterOfMistakes** [flatpak/xdg-desktop-portal#1222] [Ideas]
Suggestion

Discord can't detect games running from Steam because a flatpak app process is isolated, is the same with Lutris or Bottles.

Purpose: A Flatpak Game Portal that can be a alternative, a client can give to portal frontend the game that are running,
and a other client can too view if a gaming are running.

If some thing look this can be developed will be great not just for Discord, but for others apps that can view if a game is running, it can be not just using for simple things like a status about playing but for optimizations in applications and system side, in this case reduce animations, reduce effects UI in application or Shell and in some cases the system can modify the Governor and Energy settings if game is detected.

**Comment by Mikenux** [comment]
For game names:
- Maybe we can get this, depending on how the future Accessibility API is developed. In this case, it's just about getting the app names.
- Should this be limited to games? In other words, is a limitation to specific app categories and apps/games wanted? (certainly because, as I've seen in other issues, there are windows with passwords and developers working on secret projects).

For optimizations:
- There is the Game Mode portal.
- Missing items should be included in Game Mode, or the portal should be revised to have scope not limited to Feral Game Mode.

**Comment by razzeee** [comment]
Discord can also listen to what spotify does and display what song you are listening to, via the same/similar api.

I also think ideally it's more complex then just a window title, the steam api allows for setting some more strings and I think discord allows for that too.

So that you can not only show what they are playing, but some more context.

**Reply by Mikenux** [reply]
Ok. So the portal must know, objectively, that the game uses the Steam or Discord API. So in the portal UI we can tell the user that the game wants to share data with Steam/Discord.

**Reply by razzeee** [reply]
I would think, that steam still gets that data via their api right now, but would have to test it.

**Reply by Mikenux** [reply]
Sorry, I was thinking about inter-app communication, which may not be the case for the Steam/Discord API. The data is certainly sent to servers. In such a case, only a portal restricting network activity could be useful. There is already discussions about an on/off network portal (https://github.com/flatpak/xdg-desktop-portal/discussions/1166), but restricting network activity is not a topic for now (because they first need to know if network can be manipulated per sandbox). If this is the case, then restriction could be considered, but it will be necessary to see the cases to know if things can be easily communicated to users.

**Reply by razzeee** [reply]
I think we want to make it work in the first place, not break it more? :)

**Reply by Mikenux** [reply]
It's a privacy issue, which means asking users if they agree to share their data. There is nothing to break this kind of functionality :wink:

**Reply by razzeee** [reply]
I don't think we can reasonably expect games to implement portals. With luck we can get steam / spotify to do that.

That basically means this is a dead idea.

**Reply by Mikenux** [reply]
If the game is in a sandbox, therefore identifiable, there are two solutions:
- Have something ready to use to ask for permission at game start-up (if possible);
- Ask the application that collects this information to implement it. This means having the portal UI in the form of an allow/deny list.

However, we have to wait to know which cases the accessibility mechanism will cover.

Regarding the network, information about “Playing Now” is unlikely to be transmitted via the network. Furthermore, even if we had a network restriction, it probably wouldn't change anything since data can be transmitted via, for example, the game server which will not be blocked if the game requires Internet access (and the user allowed it).

**Reply by razzeee** [reply]
I would expect those calls to be calls to the steam sdk maybe rpc/IPC? I don't think they are actually network bound. Steam then will probably propagate them via network.

Yes, it's rpc/ipc https://partner.steamgames.com/doc/sdk/api#3

**Reply by Mikenux** [reply]
The problem is that we cannot control these APIs.

Perhaps the following can be done, knowing that this is only for standalone games, i.e. games that do not rely on a platform (e.g. Steam) to launch:
- Games use the `game` category.
- Games indicate which APIs they support: `steam`, `discord`, any others that might be added.
- The app that wants to get game data requests access to local game data via the Steam API or Discord.
- If the user agrees, the games and the requesting app can exchange data provided that the game and the app cannot delete or modify each other's files (including the app/game settings).

However, this requires either limiting the use of this portal to Steam or Discord, or relying on the portal UIs or users to tell/guess that it is Steam/Discord only (because any app can make the request).

This proposal is, of course, up for judgment (is something problematic? is something missing? is it acceptable? etc.).

**Reply by EliasTheGrandMasterOfMistakes** [reply]
About Discord, it handles the name of process, of if make more thing?  because if this use Dbus session is just allow to view pid steam process or use a other way? but about Spotify is necessary to view if is the same thing, sorry for my question

I think if a people is using Discord  i think he is according with Discord Agreement Privacy polities.

But in some cases is possible to have just a permission to allow discord handle this thing with break sandbox, but idk if is the best way.

But to have the best way looks very hardly because Discord Devs teams and Steam needs to enter in agreement about a 
possible solution for this, or in some cases.

About Spotify i think they will not help with fix this

**Reply by EliasTheGrandMasterOfMistakes** [reply]
sorry for this very confusion sugestion

**Reply by EliasTheGrandMasterOfMistakes** [reply]
Sorry for this question, in this case is better to waiting Discord Dev teams work with a proposal if they want this?.
Discord now is verified this means it's can be solved by devs team in future ?

**Reply by Mikenux** [reply]
Unfortunately, I'm not a developer, that's why I mention rules.

The way apps communicate with each other is indeed important. It would indeed be interesting to have the opinion of developers who know these APIs and how they communicate, if this needs something extra so that the relevant data is transferred via dbus instead.  Maybe the Flatpak developers can also say if this is possible or maybe allow private portals provided they follow specific rules (i.e. file protection, and also not being able to remotely control the app or let the app do the filtering itself).

Regarding the agreement policies, it is up to the app to display it before making the request, or it is a question of including a link to it in the portal.

**Reply by EliasTheGrandMasterOfMistakes** [reply]
thank you, can be good with a agreement policies
sadly is necessary to view if discord have interesting in thing like this, i heard about discord will be launch official releases
